### PR TITLE
Tpetra: Fix build warnings

### PR DESCRIPTION
**Pull request by mhoemmen**
@trilinos/tpetra @trilinos/framework 

Fix some Tpetra build warnings that block building with warnings as errors.

### CODE REVIEW

File: packages/tpetra/core/src/Tpetra_DistObject_def.hpp [hunk 1]
```diff
@@ -1856,13 +1855,7 @@ namespace Tpetra {
...
   removeEmptyProcessesInPlace (Teuchos::RCP<DistObjectType>& input)
   {
-    using Teuchos::RCP;
-    typedef typename DistObjectType::local_ordinal_type LO;
-    typedef typename DistObjectType::global_ordinal_type GO;
-    typedef typename DistObjectType::node_type NT;
-    typedef Map<LO, GO, NT> map_type;
-
-    RCP<const map_type> newMap = input->getMap ()->removeEmptyProcesses ();
+    auto newMap = input->getMap ()->removeEmptyProcesses ();
```
Comment (bartlettroscoe) [inline packages/tpetra/core/src/Tpetra_DistObject_def.hpp]: 'auto' :-)

File: packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp [hunk 2]
```diff
@@ -155,21 +155,29 @@ namespace { // (anonymous)
          * the number of replaced diagonal entries to match the
          * local number of rows.
          */
-        TEST_EQUALITY(numReplacedDiagEntries, matrix->getNodeNumRows());
+	const LO lclNumRows = static_cast<LO> (matrix->getNodeNumRows ());
+        TEST_EQUALITY(numReplacedDiagEntries, lclNumRows);
```
Comment (bartlettroscoe) [inline packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp]: Does this indentation not quite line up here?  Looks like TEST_EQUALITY() is indented one space too many.

File: packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp [hunk 3]
```diff
@@ -155,21 +155,29 @@ namespace { // (anonymous)
...
          */
 
-        RCP<vec_type> diagCopy = rcp(new vec_type(matrix->getRowMap()));
-        matrix->getLocalDiagCopy(*diagCopy);
-
-        Teuchos::ArrayRCP<const Scalar> diagCopyData = diagCopy->getData();
-
-        for (size_t i = 0; i < diagCopyData.size(); ++i)
-          TEST_EQUALITY_CONST(diagCopyData[i], rankAsScalar);
+        vec_type diagCopy (matrix->getRowMap ());
```
Comment (bartlettroscoe) [inline packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp]: deep copy of the row map?

Comment (mhoemmen) [reply packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp]: @bartlettroscoe Nope -- `getRowMap()` returns `RCP<const Map>`.

**Comment by mhoemmen** [comment]
@bartlettroscoe It's OK to approve, but if you feel better letting somebody else do it, feel free.  Thanks!

File: packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp (hunk 3 above)
Comment (bartlettroscoe) [reply packages/tpetra/core/test/CrsMatrix/CrsMatrix_ReplaceDiagonal.cpp]: Right.  I forgot Tpetra moved to shallow copy semantics.

**Pull request merged**